crates/transaction-pool/src/validate/eth.rs

```diff
@@ -297,6 +299,30 @@ where
             )
         }
 
+        if self.local_transactions_config.is_local(origin, transaction.sender_ref()) {
+            match self.tx_fee_cap {
+                Some(0) | None => {} // Skip if cap is 0 or None
+                Some(tx_fee_cap_wei) => {
+                    let gas_price = if transaction.is_dynamic_fee() {
+                        transaction.max_fee_per_gas()
+                    } else {
+                        transaction.gas_price().unwrap_or(0)
+                    };
+
+                    let max_tx_fee_wei = gas_price * (transaction.gas_limit() as u128);
+                    if max_tx_fee_wei > tx_fee_cap_wei {
+                        return TransactionValidationOutcome::Invalid(
+                            transaction,
+                            InvalidPoolTransactionError::ExceedsFeeCap {
+                                max_tx_fee_eth: max_tx_fee_wei as f64 / 1e18,
+                                tx_fee_cap_eth: tx_fee_cap_wei as f64 / 1e18
+                            },
+                        );
```

Comment:
I'd like an explainer what this checks exactly, all fee related stuff is always hard to read, so this requires docs